Decred Wallet/Features/Receive/ReceiveViewController.swift

```diff
@@ -0,0 +1,196 @@
+    }
+
+    func setupExtraUI() {
+        self.addressQRCodeImageView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.walletAddressLabel.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.selectedAccountView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.showAccountSelectorDialog)))
+    }
+
+    @objc func copyAddress() {
+        DispatchQueue.main.async {
+            UIPasteboard.general.string = self.walletAddressLabel.text!
+            Utils.showBanner(parentVC: self, type: .success, text: LocalizedStrings.walletAddrCopied)
+        }
+    }
+
+    @objc func showAccountSelectorDialog(_ sender: Any) {
+        AccountSelectorDialog.show(sender: self,
+                                 title: LocalizedStrings.receivingAccount,
+                                 selectedWallet: selectedWallet,
+                                 selectedAccount: self.selectedAccount,
+                                 callback: self.updateSelectedAccount)
+    }
+
+    private func checkSyncStatus() {
+        self.moreMenuButton.isEnabled = false
+    
+        let accountsFilterFn: (DcrlibwalletAccount) -> Bool = { $0.totalBalance > 0 || $0.name != "imported" }
+        guard let wallet = WalletLoader.shared.wallets.map({ Wallet.init($0, accountsFilterFn: accountsFilterFn) }).first,
+            let account = wallet.accounts.first else {
+                self.mainContentViewHolder.isHidden = true
+                self.syncInProgressLabel.isHidden = false
+                return
+        }
+
+        self.updateSelectedAccount(wallet.id, account)
+
+        mainContentViewHolder.isHidden = false
+        syncInProgressLabel.isHidden = true
+        self.moreMenuButton.isEnabled = true
+    }
+
+    private func generateNewAddress() {
+        self.oldAddress = self.walletAddressLabel.text!
+        self.getNextAddress()
+    }
+
+    private func updateWalletAddressAndQRCode(receiveAddress: String) {
+        DispatchQueue.main.async {
+            self.walletAddressLabel.text = receiveAddress
+            self.addressQRCodeImageView.image = self.generateQRCodeFor(
+                with: receiveAddress,
+                forImageViewFrame: self.addressQRCodeImageView.frame
+            )
+        }
+    }
+
+    @objc private func getNextAddress() {
+        if let wallet = self.selectedWallet,
+            let account = self.selectedAccount {
+                let receiveAddress = wallet.nextAddress(account.number, error: nil)
+                if self.oldAddress != receiveAddress {
+                    self.updateWalletAddressAndQRCode(receiveAddress: receiveAddress)
+                } else if receiveAddress != "" {
+                    self.getNextAddress()
+                }
+        }
+    }
+
+    private func generateQRCodeFor(with addres: String, forImageViewFrame: CGRect) -> UIImage? {
```

Comment:
```suggestion
    private func generateQRCodeImage(for address: String) -> UIImage? {
```
Don't need an imageViewFrame parameter, this function has direct access to `self.addressQRCodeImageView.frame`.